godtossing/Pos/Binary/GodTossing/Types.hs

```diff
@@ -4,35 +4,47 @@ module Pos.Binary.GodTossing.Types () where
 
 import           Universum
 
-import           Pos.Binary.Class                 (Cons (..), Field (..), deriveSimpleBi)
-import           Pos.Core.Types                   (EpochIndex, EpochOrSlot, StakeholderId)
-import           Pos.Ssc.GodTossing.Core          (CommitmentsMap, Opening, OpeningsMap,
-                                                   SharesMap, SignedCommitment,
-                                                   VssCertificate, VssCertificatesMap)
+import           Pos.Binary.Class                 (Bi (..), Cons (..), Field (..), deriveSimpleBi, encodeListLen,
+                                                   enforceSize)
+import           Pos.Core                         (EpochIndex, HasCoreConstants)
+import           Pos.Ssc.GodTossing.Core          (Opening, SignedCommitment,
+                                                   VssCertificatesMap)
 import           Pos.Ssc.GodTossing.Genesis.Types (GenesisGtData (..))
 import           Pos.Ssc.GodTossing.Types         (GtGlobalState (..),
                                                    GtSecretStorage (..))
 import           Pos.Ssc.GodTossing.VssCertData   (VssCertData (..))
 
-deriveSimpleBi ''VssCertData [
-    Cons 'VssCertData [
-        Field [| lastKnownEoS :: EpochOrSlot                       |],
-        Field [| certs        :: VssCertificatesMap                |],
-        Field [| whenInsMap   :: HashMap StakeholderId EpochOrSlot |],
-        Field [| whenInsSet   :: Set (EpochOrSlot, StakeholderId)  |],
-        Field [| whenExpire   :: Set (EpochOrSlot, StakeholderId)  |],
-        Field [| expiredCerts :: Set (EpochOrSlot, (StakeholderId,
-                                                      EpochOrSlot,
-                                                      VssCertificate)) |]
-    ]]
+instance HasCoreConstants => Bi VssCertData where
```

Comment:
mmmmm, nice plan.
what about other constraints? with non empty type-parametes.
I will try now, but it will look like kostil for `HasCoreConstants`